SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -108,12 +108,7 @@ public bool Execute(ProcessedArgs args, ILogger logger)
 
         private bool FetchArgumentsAndRulesets(ProcessedArgs args, AnalysisConfig config, ILogger logger)
         {
-            string hostUrl = GetServer(args);
-            if (String.IsNullOrEmpty(hostUrl))
-            {
-                logger.LogError(Resources.ERROR_NoHostUrl);
-                return false;
-            }
```

Comment:
This logic was in the wrong place: the argument processing and validation should be done in the ArgumentProcessor, not in this class. I moved the code and fixed up the unit tests.